src/index.js

```diff
@@ -50,36 +43,28 @@ SavitzkyGolay.prototype.calc = function (options) {
         }
     }
     else {
-        var J = new Array(options.windowSize);
+        var J = Matrix.ones(options.windowSize, options.polynomial + 1);
+        var inic = -(options.windowSize - 1) / 2;
         for (var i = 0; i < J.length; i++) {
-            J[i] = new Array(options.polynomial + 1);
-            var inic = -(options.windowSize - 1) / 2;
             for (var j = 0; j < J[i].length; j++) {
-                if ((inic + 1 === 0) && (j === 0))
-                    J[i][j] = 1;
-                else
+                if ((inic + 1 !== 0) || (j !== 0))
                     J[i][j] = Math.pow((inic + i), j);
             }
         }
-        var Jmatrix = new matrix(J);
-        var Jtranspose = Jmatrix.transpose();
-        var Jinv = (Jtranspose.mmul(Jmatrix)).inverse();
+        var Jtranspose = J.transpose();
+        var Jinv = (Jtranspose.mmul(J)).inverse();
         C = Jinv.mmul(Jtranspose);
         C = C[options.derivative];
         norm = 1;
     }
-    for (var k = Math.ceil(options.windowSize / 2); k < (ans.length - Math.floor(options.windowSize / 2)); k++) {
+    var det = norm * Math.pow(h, options.derivative);
```

Comment:
The norm value is just because they want integer values in the tables. If you use the normal code to generate the weight vectors, you will get a real value, and on those cases the norm value is always 1. There is a second algorithm that stores the numerator and denominator separately, and at the end can choose the appropriated norm value to let all your weights as integers.